label_studio/data_import/api.py

```diff
@@ -327,7 +327,11 @@ class ImportPredictionsAPI(generics.CreateAPIView):
     def create(self, request, *args, **kwargs):
         # check project permissions
         project = self.get_object()
-        tasks_ids = set(Task.objects.filter(project=project).values_list('id', flat=True))
+
+        tasks_ids_and_project_ids = set(Task.objects.filter(project=project).values_list('id', 'project_id'))
+        tasks_ids = {task_id_and_project_id[0] for task_id_and_project_id in tasks_ids_and_project_ids}
+        task_to_project_mapping = {task_id: project_id for task_id, project_id in tasks_ids_and_project_ids}
```

Comment:
```suggestion
```

We reasonably could also remove this and just grab the direct project id since we already have it.

Reply:
💯